src/application/predictor.hpp

```diff
@@ -251,10 +251,9 @@ class Predictor {
 
  private:
   void CopyToPredictBuffer(double* pred_buf, const std::vector<std::pair<int, double>>& features) {
-    int loop_size = static_cast<int>(features.size());
-    for (int i = 0; i < loop_size; ++i) {
-      if (features[i].first < num_feature_) {
-        pred_buf[features[i].first] = features[i].second;
+    for (const auto &feature: features) {
```

Comment:
To fix and pass CI tests.
```suggestion
    for (const auto &feature : features) {
```